### ISSUE TITLE
Optimise Mapbox#setConnected

### ISSUE THREAD

**Issue by tobrun** [mapbox/mapbox-gl-native#15698]
The `Mapbox#setConnected API` was introduced for embedded devices that didn't support a connectivity state through Android internal connectivity change events delivered through intents. These developers call `Mapbox#setConnected(true)` to always have the device connected and load tiles. With iOS introducing a new API to pause network requests, we should optimize our setup to match this.

**Comment by pengdev** [comment]
@tobrun Should this `setConnected` be located in `Mapbox` class or `MapboxMap` class? It seems it should be in the `Mapbox` class but just want to confirm.

**Comment by tobrun** [comment]
Correct, we need to update the existing `Mapbox#setConnected` method to call into the c++ code so we can do something as `mbgl::NetworkStatus::Set(NetworkStatus::Status::Offline).`
